src/components/integrated/OakLessonNavItem/OakLessonNavItem.tsx

```diff
@@ -0,0 +1,304 @@
+import React, { ComponentPropsWithoutRef, ElementType } from "react";
+import styled, { css } from "styled-components";
+
+import {
+  OakBox,
+  OakFlex,
+  OakIcon,
+  OakIconName,
+  OakSpan,
+} from "@/components/base";
+import { parseColor } from "@/styles/helpers/parseColor";
+import { OakRoundIcon } from "@/components/ui";
+import { parseColorFilter } from "@/styles/helpers/parseColorFilter";
+import { parseSpacing } from "@/styles/helpers/parseSpacing";
+import { parseDropShadow } from "@/styles/helpers/parseDropShadow";
+import { OakCombinedColorToken } from "@/styles";
+
+type LessonSectionName = "intro" | "starter-quiz" | "video" | "exit-quiz";
+
+type BaseOakLessonNavItemProps<C extends ElementType> = {
+  as?: C;
+  /**
+   * Present the section as disabled.
+   *
+   * If this is a button the `disabled` prop should also be set.
+   * For a link the anchor should be replaced with a non interactive element
+   */
+  isDisabled?: boolean;
+} & ComponentPropsWithoutRef<C>;
+
+type QuizSectionProps = {
+  lessonSectionName: "starter-quiz" | "exit-quiz";
+  /**
+   * The number of questions in the quiz
+   */
+  numQuestions: number;
+  /**
+   * The number of questions answered correctly
+   */
+  grade: number;
+};
+
+type VideoSectionProps = {
+  lessonSectionName: "video";
+  /**
+   * The length of the video in minutes
+   */
+  videoLength: number;
+};
+
+type IntroSectionProps = {
+  lessonSectionName: "intro";
+};
+
+type SectionProps = {
+  /**
+   * Denotes the progress in the lesson section
+   */
+  progress: "not-started" | "in-progress" | "complete";
+} & (IntroSectionProps | QuizSectionProps | VideoSectionProps);
+
+export type OakLessonNavItemProps<C extends ElementType> =
+  BaseOakLessonNavItemProps<C> & SectionProps;
+
+const StyledLabel = styled(OakBox)``;
+
+const StyledRoundIcon = styled(OakRoundIcon)<{
+  $isDisabled?: boolean;
+}>`
+  width: ${parseSpacing("all-spacing-8")};
+  height: ${parseSpacing("all-spacing-8")};
+  padding: 0;
+
+  background: transparent;
+
+  img {
+    filter: ${(props) =>
+      parseColorFilter(props.$isDisabled ? "icon-disabled" : "icon-inverted")};
+  }
+`;
+
+const StyledLessonNavItem = styled(OakFlex)<{ $isDisabled?: boolean }>`
+  outline: none;
+  text-align: initial;
+
+  &:focus-visible {
+    box-shadow: ${parseDropShadow("drop-shadow-centered-yellow")},
+      ${parseDropShadow("drop-shadow-centered-grey")};
+  }
+
+  ${(props) => props.$isDisabled && "cursor: default"}
+
+  ${(props) =>
+    !props.$isDisabled &&
+    css`
+      cursor: pointer;
+
+      &:hover,
+      &:active {
+        ${StyledLabel} {
+          text-decoration: underline;
+        }
+
+        ${StyledRoundIcon} {
```

Comment:
This breaks encapsulation of component styles in `OakRoundIcon` and I loath it. My only other option was to write more JavaScript and use pointer events to flip some props. I'd loath that even more.

Reply:
Not sure what you mean by inherit here?

Reply:
I will try to explain to you on a phone call and see if I'm talking rubbish.